internal/indexcoord/index_coord.go

```diff
@@ -538,6 +538,68 @@ func (i *IndexCoord) GetSegmentIndexState(ctx context.Context, req *indexpb.GetS
 	return ret, nil
 }
 
+// CompleteIndexInfo get the building index progress and index state
+func (i *IndexCoord) completeIndexInfo(ctx context.Context, indexInfo *indexpb.IndexInfo) error {
+	collectionID := indexInfo.CollectionID
+	indexName := indexInfo.IndexName
+	log.Info("IndexCoord completeIndexInfo", zap.Int64("collID", collectionID),
+		zap.String("indexName", indexName))
+	flushSegments, err := i.dataCoordClient.GetFlushedSegments(ctx, &datapb.GetFlushedSegmentsRequest{
+		CollectionID: collectionID,
+		PartitionID:  -1,
+	})
+	if err != nil {
+		return err
+	}
+
+	resp, err := i.dataCoordClient.GetSegmentInfo(ctx, &datapb.GetSegmentInfoRequest{
+		SegmentIDs: flushSegments.Segments,
```

Comment:
Can we combine these two RPCs `GetFlushedSegments` and `GetSegmentInfo` into one `GetFlushedSegmentsInfo`? I was wondering if there are any chances that `flushSegments.Segments` will contain compacted segments.

Reply:
These code are same as the `GetIndexBuildProgress`. I think combining the RPCs are unreasonable, because there is somewhere else calling one of the two RPCs, not needing all the information.

Reply:
I don't mean we only need one RPC. I just want to say that we should care of this case:
Between the gap of `GetFlushedSegments` and `GetSegmentInfo`, the segments returned by `GetFlushedSegments` may be compacted.